File: 1859. Sorting the Sentence/Solution.cpp
#include <sstream>
#include <string>
#include <vector>
using namespace std;

class Solution
{
public:
	string sortSentence(string s)
	{
		vector<string> vs(9);
		istringstream in(s);
		string res, word;
		while (in >> word)
			vs[word.back() - '1'] = word.substr(0, word.size() - 1);
		res = vs[0];
		for (size_t i = 1; i < vs.size() && vs[i].size(); i++)
			res += " " + vs[i];
		return res;
	}
};

File: 1859. Sorting the Sentence/Solution.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String sortSentence(String s) {
        String[] arr = s.split(" ");
        Arrays.sort(arr, Comparator.comparingInt(a -> a.charAt(a.length() - 1)));
        StringBuilder sb = new StringBuilder();
        for (String word : arr) {
            sb.append(word, 0, word.length() - 1).append(" ");
        }
        return sb.substring(0, sb.length() - 1);
    }
}
